metadata/src/main/java/org/apache/kafka/controller/metrics/QuorumControllerMetrics.java

```diff
@@ -148,6 +159,41 @@ public Long value() {
         }));
     }
 
+    public void addTimeSinceLastHeartbeatMetric(int brokerId) {
+        brokerContactTimesMs.put(brokerId, new AtomicLong(time.milliseconds()));
+        registry.ifPresent(r -> r.newGauge(
+            getBrokerIdTagMetricName(
+                "KafkaController",
+                TIME_SINCE_LAST_HEARTBEAT_RECEIVED_METRIC_NAME,
+                brokerId
+            ),
+            new Gauge<Integer>() {
+                @Override
+                public Integer value() {
+                    return timeSinceLastHeartbeatMs(brokerId);
+                }
+            }
+        ));
+    }
+
+    public void removeTimeSinceLastHeartbeatMetric(int brokerId) {
+        brokerContactTimesMs.remove(brokerId);
+        registry.ifPresent(r -> r.removeMetric(
```

Comment:
we need to remove the metric first to avoid having a race condition here, right? since otherwise it could be locklessly accessed after being removed from `brokerContactTimesMs` but before the metric is unregistered.